src/server/.tests/ws-routes/controllers/Friends.spec.js

```diff
@@ -0,0 +1,497 @@
+		} 
+		suite.client = {}
+		suite.userModelMock = {}
+
+	})
+
+	afterEach(done => {
+		suite.server.close(done)
+		if(_.isFunction(suite.client.disconnect)) suite.client.disconnect()
+	})
+
+	describe('#Connection', () => {
+		beforeEach(() => {
+			suite.friendsInstance = new FriendsProvider({
+				UserModel: sinon.stub()
+			})
+		})
+
+		it('should add new element to user-socket map', done => {
+			//given
+			suite.server.on(CLIENT_EVENTS.CONNECTION, socket => {
+				suite.friendsInstance.connection(socket, suite.connectionsMock)
+				then()
+			})
+
+			//when
+			suite.client = socketClient.connect(SERVER_URL, SOCKET_OPTIONS)
+
+			//then
+			function then() {
+				const hasSocketStored = suite.connectionsMock.usersToConnectionsMap.has(suite.DUMMY_USERNAME)
+				assert.isTrue(hasSocketStored)
+				done()
+			}
+		})
+
+	})
+
+	describe('#Friends list', () => {
+		beforeEach(() => {
+			suite.userModelMock = {
+				findOne: sinon.stub()
+			}
+			suite.friendsInstance = new FriendsProvider({
+				UserModel: suite.userModelMock
+			})
+		})
+
+		it('should sent event to client who requested for it', done => {
+			//given
+			const QUERY_FEEDBACK_MOCK = []
+			const queryResultMock = {
+				exec: sinon.stub().resolves(QUERY_FEEDBACK_MOCK)
+			}
+			suite.userModelMock.findOne.returns(queryResultMock)
+
+			suite.server.on(CLIENT_EVENTS.CONNECTION, connection => {
+				connection.on(CLIENT_EVENTS.FRIENDS_LIST, data => {
+					suite.emitSpy = sinon.spy(connection, 'emit')
+					return suite.friendsInstance.friendsList(data, connection)
+						.then(() => {
+							//then
+							sinon.assert.calledOnce(suite.emitSpy)
+							done()
+						})
+				})
+			})
+
+			//when
+			suite.client = socketClient.connect(SERVER_URL, SOCKET_OPTIONS)
+			suite.client.emit(CLIENT_EVENTS.FRIENDS_LIST)
+		})
+
+		it('should emit friendsList event type with friends list attached to it', done => {
+			//given
+			const QUERY_FEEDBACK_MOCK = {
+				username: suite.DUMMY_USERNAME,
+				friends: [
+					{
+						$id: 'dummyId1',
+						username: 'dummyFriend1'
+					},
+					{
+						$id: 'dummyId2',
+						username: 'dummyFriend2'
+					}
+				]
+			}
+
+			const queryResultMock = {
+				exec: sinon.stub().resolves(QUERY_FEEDBACK_MOCK)
+			}
+
+			suite.userModelMock.findOne.returns(queryResultMock)
+
+			suite.server.on(CLIENT_EVENTS.CONNECTION, connection => {
+				suite.emitSpy = sinon.spy(connection, 'emit')
+				connection.on(CLIENT_EVENTS.FRIENDS_LIST, data => {
+					return suite.friendsInstance.friendsList(data, connection)
+						.then(() => {
+
+							//then
+							const expectedPayload = QUERY_FEEDBACK_MOCK.friends
+							sinon.assert.calledWith(suite.emitSpy, CLIENT_EVENTS.FRIENDS_LIST, expectedPayload)
+							done()
+						})
+				})
+			})
+
+			//when
+			suite.client = socketClient.connect(SERVER_URL, SOCKET_OPTIONS)
+			suite.client.emit(CLIENT_EVENTS.FRIENDS_LIST)
+		})
+	})
+
+	describe('#Notifications', () => {
```

Comment:
Should notifications be part of `Friends` namespace?

Reply:
ok.